### PR TITLE
Clean up and separate out tests

### LINKED ISSUE #21: Clean up ProjectTests
Make them more specific to what they're testing.

### PR DESCRIPTION
**Pull request by loosebazooka**
fixes #21

### CODE REVIEW

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.maven.it.VerificationException;
+import org.apache.maven.it.Verifier;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.matchers.JUnitMatchers;
+import org.junit.rules.TemporaryFolder;
+
+/** Test for ClientLibsMojo. */
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java]: Is this comment necessary?

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    File pom = new File(dir, "pom.xml");
+    String pomContents = FileUtils.fileRead(pom);
+    pomContents = pomContents.replaceAll("@@PluginVersion@@", pluginVersion);
+    FileUtils.fileWrite(pom, pomContents);
+
+    return dir;
+  }
+
+  // inject a endpoints plugin configuration into the pom.xml
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java]: inject **an** endpoints

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  @Test
+  public void testApplicationId() throws IOException, VerificationException, XmlPullParserException {
+    File testDir = new TestProject(tmpDir.getRoot(), "/projects/server").applicationId("maven-test").build();
+
+    Verifier verifier = new Verifier(testDir.getAbsolutePath());
+    verifier.executeGoal("endpoints-framework:clientLibs");
+    verifier.verifyErrorFreeLog();
+
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java]: `verifier.assertFilePresent(CLIENT_LIB_PATH);`?

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  @Test
+  public void testHostname() throws IOException, VerificationException, XmlPullParserException {
+    File testDir = new TestProject(tmpDir.getRoot(), "/projects/server")
+        .configuration("<configuration><hostname>my.hostname.com</hostname></configuration>")
+        .build();
+
+    Verifier verifier = new Verifier(testDir.getAbsolutePath());
+    verifier.executeGoal("endpoints-framework:clientLibs");
+    verifier.verifyErrorFreeLog();
+
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java]: `verifier.assertFilePresent(CLIENT_LIB_PATH);`?

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/DiscoveryDocsMojoTest.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.matchers.JUnitMatchers;
+import org.junit.rules.TemporaryFolder;
+
+public class DiscoveryDocsMojoTest {
+
+  private static final String DEFAULT_HOSTNAME = "myapi.appspot.com";
+  private static final String DEFAULT_URL = "https://" + DEFAULT_HOSTNAME + "/_ah/api/";
+  private static final String DISC_DOC_PATH = "target/discovery-docs/testApi-v1-rest.discovery";
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/DiscoveryDocsMojoTest.java]: `DISCOVERY_DOC_PATH`

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/DiscoveryDocsMojoTest.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  @Test
+  public void testApplicationId() throws IOException, VerificationException, XmlPullParserException {
+    File testDir = new TestProject(tmpDir.getRoot(), "/projects/server")
+        .applicationId("maven-test")
+        .build();
+
+    Verifier verifier = new Verifier(testDir.getAbsolutePath());
+    verifier.executeGoal("endpoints-framework:discoveryDocs");
+    verifier.verifyErrorFreeLog();
+
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/DiscoveryDocsMojoTest.java]: `verifier.assertFilePresent(DISC_DOC_PATH);`?

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/DiscoveryDocsMojoTest.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  @Test
+  public void testHostname() throws IOException, VerificationException, XmlPullParserException {
+    File testDir = new TestProject(tmpDir.getRoot(), "/projects/server")
+        .configuration("<configuration><hostname>my.hostname.com</hostname></configuration>")
+        .build();
+
+    Verifier verifier = new Verifier(testDir.getAbsolutePath());
+    verifier.executeGoal("endpoints-framework:discoveryDocs");
+    verifier.verifyErrorFreeLog();
+
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/DiscoveryDocsMojoTest.java]: `verifier.assertFilePresent(DISC_DOC_PATH);`?

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java (hunk 1 above)
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java]: checkstyle.

**Comment by loosebazooka** [comment]
updated.

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java (hunk 1 above)
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/ClientLibsMojoTest.java]: I thought checkstyle isn't on for `src/test`.

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  public TestProject applicationId(String applicationId) {
+    this.application = applicationId;
+    return this;
+  }
+
+  public File build() throws IOException, VerificationException, XmlPullParserException {
+    prepareEnvironment();
+    File root = build(projectPath);
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java]: `projectRootDirectory` or `projectDirectory`?

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+      injectApplicationId(root, application);
+    }
+    if (configuration != null) {
+      injectConfiguration(root, configuration);
+    }
+    return root;
+  }
+
+  private File build(String path) throws IOException {
+    File dir = ResourceExtractor
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java]: `projectRootDirectory` or `projectDirectory`?

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    File pom = new File(dir, "pom.xml");
+    String pomContents = FileUtils.fileRead(pom);
+    pomContents = pomContents.replaceAll("@@PluginVersion@@", pluginVersion);
+    FileUtils.fileWrite(pom, pomContents);
+
+    return dir;
+  }
+
+  // inject an endpoints plugin configuration into the pom.xml
+  private void injectConfiguration(File root, String configuration) throws IOException {
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java]: `projectRootDirectory` or `projectDirectory`?

File: src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+    File pom = new File(root, "pom.xml");
+    String pomContents = FileUtils.fileRead(pom);
+    pomContents = pomContents.replaceAll("<!--endpoints-plugin-configuration-->", configuration);
+    FileUtils.fileWrite(pom, pomContents);
+  }
+
+  // inject an application tag into the appengine-web.xml
+  private void injectApplicationId(File root, String application) throws IOException {
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/maven/endpoints/framework/TestProject.java]: `projectRootDirectory` or `projectDirectory`?

**Pull request merged**